NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1342,370 @@ def __init__(self, inputfilename,
-                    if iEvent < n_sub_events - 1:
-                        iStop = iSplit[iEvent] + 1
-                    indices = start_times_sort[iStart: iStop]
-                    if n_sub_events > 1:
-                        tmp = ""
-                        for start_time in start_times[indices]:
-                            tmp += f"{start_time/units.ns:.0f}, "
-                        tmp = tmp[:-2] + " ns"
-                        logger.info(f"creating event {iEvent} of event group {self._event_group_id} ranging rom {iStart} to {iStop} with indices {indices} corresponding to signal times of {tmp}")
-                    self._evt = NuRadioReco.framework.event.Event(self._event_group_id, iEvent)  # create new event
-
-                    if particle_mode:
-                        # add MC particles that belong to this (sub) event to event structure
-                        # add only primary for now, since full interaction chain is not typically in the input hdf5s
-                        self._evt.add_particle(self.primary)
-                    # copy over generator information from temporary event to event
-                    self._evt._generator_info = self._generator_info
-
-                    self._station = NuRadioReco.framework.station.Station(self._station_id)
-                    sim_station = NuRadioReco.framework.sim_station.SimStation(self._station_id)
-                    sim_station.set_is_neutrino()
-                    tmp_sim_station = tmp_station.get_sim_station()
-                    self._shower_ids_of_sub_event = []
-                    for iCh in indices:
-                        ch_uid = channel_identifiers[iCh]
-                        shower_id = ch_uid[1]
-                        if shower_id not in self._shower_ids_of_sub_event:
-                            self._shower_ids_of_sub_event.append(shower_id)
-                        sim_station.add_channel(tmp_sim_station.get_channel(ch_uid))
-                        efield_uid = ([ch_uid[0]], ch_uid[1], ch_uid[2])  # the efield unique identifier has as first parameter an array of the channels it is valid for
-                        for efield in tmp_sim_station.get_electric_fields():
-                            if efield.get_unique_identifier() == efield_uid:
-                                sim_station.add_electric_field(efield)
+                                channel = NuRadioReco.framework.channel.Channel(channel_id)
+                                n_samples = int(round(self._det.get_number_of_samples(sid, channel_id)) * self._config['sampling_rate'] / self._det.get_sampling_frequency(sid, channel_id))
+                                channel.set_trace(np.zeros(n_samples), self._config['sampling_rate'])
+                                # we need to use any other channel to get the correct trace_start_time. All channels have the same start time at the end
+                                # of the simulation.
```

Comment:
Is that still necessary? You are adding empty channels in the previous for loop (might be did not test it).

Reply:
I'm pretty sure that this is still needed because the previous loop will skip channels that don't have RT solutions for example.